### DISCUSSION TITLE
How to list pages in /pages?

### DISCUSSION THREAD

**Discussion by pochat** [nuxt/nuxt#24983] [Questions] [answered]
Hi all,

I hope you are well.

I have pages in the /pages directory, and I would like to use a v-for to show all these pages. However, I can't find any documentation on how to achieve this.

For example, my pages directory is:
-/pages
--index.vue
--page1.vue
--page2vue
--page3vue

A component that loops through the /pages directory to create a listing of these pages.

Any ideas?

Thank you!

PS. I am using nuxt 3. I found that this is possible with Nuxt Content. https://content.nuxt.com/components/content-list

**Comment by masyoudi** [comment]
if you want to show all pages why not just create component, and then show them using v-for

**Reply by pochat** [reply]
hi @masyoudi 

I hope you’re well. 

I’m a beginner. Would you mind showing me how?

thank you for your time.

**Comment by puzzle-it-nu** [comment, marked as answer]
Hi,

Since Nuxt /pages uses vue-router you can find the needed info starting here:
https://nuxt.com/docs/api/composables/use-router and then goto  https://router.vuejs.org/api/interfaces/Router.html#Properties-currentRoute to read about getRoutes() which will give you an object with all routes in it.

**Comment by pochat** [comment]
Hi @puzzle-it-nu,

I hope you are well and thank you for your help.

You gave me the answer on the documentation, and i was able to come up with a good first version, which I will continue to polish.

Thank you and happy new year.

`<script setup>

  const router = useRouter()
  const getPages = router.getRoutes()
  console.log("getPages")
  console.log(getPages)

</script>

<template>
  <div v-for="(pages, index) in getPages" :key="index">
    <p>-----------------------</p>
    <h2>{{ pages.name }}</h2>
    <h2>{{ pages.path }}</h2>
    <NuxtLink :to="pages.path">Page name is: {{ pages.name }}</NuxtLink>

    <p>-----------------------</p>
    <NuxtLink :to="pages.path">Page name is: {{ $t('pages.name') }}</NuxtLink>
  </div>

</template>
`
